platforms/software/ivy/src/main/java/org/gradle/api/publish/ivy/tasks/GenerateIvyDescriptor.java

```diff
@@ -41,23 +46,16 @@
 @UntrackedTask(because = "Gradle doesn't understand the data structures")
 public abstract class GenerateIvyDescriptor extends DefaultTask {
 
-    private Transient.Var<IvyModuleDescriptorSpec> descriptor = Transient.varOf();
-    private final Cached<IvyDescriptorFileGenerator.DescriptorFileSpec> ivyDescriptorSpec = Cached.of(this::computeIvyDescriptorFileSpec);
-
-    private Object destination;
-
-    @Inject
-    protected PathToFileResolver getFileResolver() {
-        throw new UnsupportedOperationException();
-    }
+    private final Transient.Var<IvyModuleDescriptorSpec> descriptor = Transient.varOf();
+    private final Provider<IvyDescriptorFileGenerator.DescriptorFileSpec> ivyDescriptorSpec = getProject().provider(() ->
```

Comment:
Is there a reason this was changed to a `Provider` instead of using `Cached`? They seem pretty equivalent here